### ISSUE TITLE
[deeplab] regarding the input for eval.py

### ISSUE THREAD

**Issue by surfreta** [tensorflow/models#4280]
In `deeplab`, it includes `eval.py`, is that for prediction purposes? After reading the code, it seems that it uses the same input data path storing the generated `tfrecord `file as used by `train.py`. Is that true?

**Comment by BlueCrow1991** [comment]
The eval.py is used to validate the model you have trained, and outout the mIOU, so you should use another set with it.

**Comment by wenouyang** [comment]
Hi BlueCrow1991, in eval.py, it calls 

```
dataset = segmentation_dataset.get_dataset(
--
FLAGS.dataset, FLAGS.eval_split, dataset_dir=FLAGS.dataset_dir)


```
The `get_dataset `function needs to use `DatasetDescriptor` defined in` segmentation_dataset.py`. which requires information of `splits_to_sizes`.  If I use new data set, does that mean I need to setup train and val again? My understanding is that `eval.py` only needs to validate the model, why I still need the train data set?

**Comment by surfreta** [comment]
Hi BlueCrow1991,  

Thanks for the response.

In `eval.py`, it calls
```
dataset = segmentation_dataset.get_dataset(
--
FLAGS.dataset, FLAGS.eval_split, dataset_dir=FLAGS.dataset_dir)

```

The `get_datasetfunction `needs to use `DatasetDescriptor `defined in `segmentation_dataset.py`. which requires information of `splits_to_sizes`. If I use new data set, does that mean I need to setup` train `and `val `again? My understanding is that` eval.py `only needs to validate the model, why I still need the` train` data set?

**Comment by cmbowyer13** [comment]
I would like to know how to access the ground truth boxes so I can use the metrics.py after the training has completed using train.py. Right now that is kind of unclear , not well commented or an example I can find of how to run eval.py from the command line. When I ran eval.py as such: "python3 eval.py --logtostderr --checkpoint_dir=object_detection/Object-Detection/training4/ --eval_dir=object_detection/Object-Detection/eval4 --pipeline_config_path=object_detection/Object-Detection/training4/ssd_mobilenet_v1_pets.config" and inspected in tensorboard. it only drew one point on the graph rather than draw the whole curve for the whole training process. I would like to see the whole curve in tensorboard like I can see the loss, but I would like to know how to access the data offline so I can use metrics.py script after I do inferencing and inspect how well my model is performing.

**Comment by BlueCrow1991** [comment]
@surfreta 
You don't need to prepare new train set. Because train set and val set both are splits of one dataset, so you just need to define them in the segmentation_dataset.py for once.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
